Design updates for Embedded Components

dmallory42 opened the pull request:
Fixes #9360

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

Note: we should hold off on merging this until the `dev/include-connect-js` PR is merged, and then target development.

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

This PR contains updates to the embedded onboarding flow to match the i4 designs: xO8YdpKfaVoP2d8NvweZ5I-fi-1095_43454

* Adds a border under the navbar.
* Updates the color and text of the button text on the nav bar to use theme colors.
* Update to the logo.
* Add left and right padding to add more separation
* Update the Stripe modal to use the Woo purple

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Launch a new site with this branch using Jurassic Ninja ([quick link](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/i4-design-updates))
* Go to the WCPay Dev Tools on the new JN site and enable the `Embedded KYC` checkbox, then save the change.
* Start a new onboarding. You should see the Embedded KYC as part of the MOX flow.
* Verify the page looks as expected, and the designs match the figma.

Screenshots:

<img width="1506" alt="Screenshot 2024-09-04 at 16 22 46" src="https://github.com/user-attachments/assets/6a0a4ede-ecaa-4c98-a84f-e043435d1142">

<img width="1506" alt="Screenshot 2024-09-04 at 16 22 54" src="https://github.com/user-attachments/assets/6e5d9ce6-2f7b-4da5-a966-c1e8e7804569">

<img width="1499" alt="Screenshot 2024-09-04 at 16 24 01" src="https://github.com/user-attachments/assets/b5aa9183-7d53-4648-b0d1-54653eb300c0">


-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

dmallory42 committed:
WIP: Adding connect.js library for Stripe.

dmallory42 committed:
WIP: Adding JS resolvers for component.

dmallory42 committed:
Merge remote-tracking branch 'origin/develop' into dev/include-connect-js

dmallory42 committed:
Updates to the onboarding logic for embedded onboarding.

dmallory42 committed:
updates to the onboarding.

dmallory42 committed:
Removing the test publishable key which was originally from Stripe docs (not a real key) to avoid confusion.

oaratovskyi committed:
WIP

oaratovskyi committed:
Change hooks call to useApiFetch call

oaratovskyi committed:
Changes to make the functionality working

oaratovskyi committed:
label TODOs better

oaratovskyi committed:
Remove not needed hooks

oaratovskyi committed:
Support publishableKey in `GET /onboarding/session`

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

oaratovskyi committed:
Implemented PO and passing onboarding params to the server

oaratovskyi committed:
Merge with Dan

oaratovskyi committed:
REmove not needed test

oaratovskyi committed:
Small refactoring

oaratovskyi committed:
Fix unit tests

oaratovskyi committed:
Merge branch 'develop' into dev/include-connect-js

oaratovskyi committed:
Add changelog

oaratovskyi committed:
Remove not needed function

oaratovskyi committed:
Add loader

oaratovskyi committed:
Add on load error handler

oaratovskyi committed:
Add styling

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Adding finalization logic and updating terminology to use american english.

oaratovskyi committed:
Extract styles to separate component

oaratovskyi committed:
Add localization

oaratovskyi committed:
Fix psalm errors

oaratovskyi committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Logic updates.

dmallory42 committed:
Remove heading/subheading on embedded form.

dmallory42 committed:
Some fixes.

dmallory42 committed:
Updates to the logic.

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Updating to hide the back button once the embedded onboarding is reached.

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Update to use american english

Co-authored-by: Oleksandr Aratovskyi <79862886+oaratovskyi@users.noreply.github.com>

dmallory42 committed:
Update to use american english

Co-authored-by: Oleksandr Aratovskyi <79862886+oaratovskyi@users.noreply.github.com>

oaratovskyi committed:
Small fixes (order the dependencies and add locale only if account session is present)

oaratovskyi committed:
Update to use american eng

dmallory42 committed:
Updates relating to passing promos to/from server.

dmallory42 committed:
Add source/from args.

dmallory42 committed:
Update variable names.

dmallory42 committed:
Add handling for error responses.

dmallory42 committed:
Add handling for error responses.

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

oaratovskyi committed:
Simple embedded onboarding test

oaratovskyi committed:
Add success case of embedded onboarding

oaratovskyi committed:
Small tweak

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Design updates to match the i4 designs.

dmallory42 committed:
Design updates.

dmallory42 commented:
cc. @orcunomattic @timmy5685 , would appreciate a quick check at the screenshots here to ensure they are in line with the expected desgin!

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9381` or branch name `dev/i4-design-updates` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/i4-design-updates) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 2a7be621c42d4b4c95b7b3742292301161f1d6e7
- Build time: 2024-09-19 09:43:40 UTC

_Note: the build is updated when a new commit is pushed to this PR._

oaratovskyi commented:
@dmallory42 I'll review this one after design / product review JFYI

mordeth commented on client/onboarding/style.scss:
```diff
@@ -27,10 +30,12 @@ body.wcpay-onboarding__body {
 					display: flex;
 					align-items: center;
 					padding: $gap-large;
+					color: #0073aaff;
```
That's not a valid HEX color :)

dmallory42 replied:
Thanks, I'm not sure where that color came from. I'm now using the color defined in the Figma.

mordeth commented on client/onboarding/style.scss:
```diff
@@ -27,10 +30,12 @@ body.wcpay-onboarding__body {
 
 					.gridicons-chevron-left {
 						margin-right: 2px;
+						fill: #0073aaff;
```
Similar to the previous comment, this is not a valid HEX color.

dmallory42 replied:
Same as above :)

mordeth reviewed (changes requested):
Thank you for working on this Dan!

I’ve included some comments suggesting a few minor improvements.

Additionally, `Icon` and `closeSmall` are no longer used, perhaps we should consider deleting them?

dmallory42 committed:
Updates to the nav button colors.

dmallory42 commented:
Thank you for the review, @mordeth! I've addressed your suggestions and removed the unused imports. If you have time for another look, it's appreciated!

dmallory42 committed:
Updating to use theme colors.

dmallory42 committed:
Remove unneeded import.

mordeth reviewed (approved):
It looks great after the changes. Excellent work!

oaratovskyi committed:
Update continue onboarding to use embedded component (#9380)

Co-authored-by: oaratovskyi <oleksandr.aratovskyi@automattic.com>
Co-authored-by: Daniel Mallory <daniel.mallory@automattic.com>
Co-authored-by: Vlad Olaru <vlad@pixelgrade.com>

dmallory42 committed:
Merge branch 'develop' into dev/include-connect-js

dmallory42 committed:
Update to the styling file to use the Woo Purple.

oaratovskyi committed:
Merge branch 'dev/include-connect-js' into dev/i4-design-updates

oaratovskyi committed:
Make changes in continue kyc page

oaratovskyi reviewed (approved):
@dmallory42 I've modified logo and the cancel button for the continue KYC page as well! Tried to refactor it, but it's tricky, so a bit of code duplication.

timmy5685 commented:
@dmallory42 - i don't know if this is related but when I create a store using the JN link above - the connect page is showing the sandbox mode notification at the top of the page before I do anything.

![screencapture-devotedly-additional-squirrel-jurassic-ninja-wp-admin-admin-php-2024-09-06-16_18_35](https://github.com/user-attachments/assets/c3f88a2a-1fb7-4606-b714-38dd657c0e08)

Is that expected? The Stripe flow is also showing test mode - so maybe it's the way you set the JN to work?

Aside from that - when I try to run through the flow and complete the Stripe OTP step - I get a session expired message and can't proceed. When I click add information again the window pops for a second and then takes me right back here.

<img width="1507" alt="Screenshot 2024-09-06 at 4 25 35 PM" src="https://github.com/user-attachments/assets/639c6329-c4c9-4a55-b8ec-1d07b57d950c">

mordeth commented:
>i don't know if this is related but when I create a store using the JN link above - the connect page is showing the sandbox mode notification at the top of the page before I do anything.

mordeth commented:
>  i don't know if this is related but when I create a store using the JN link above - the connect page is showing the sandbox mode notification at the top of the page before I do anything.

> Is that expected? The Stripe flow is also showing test mode - so maybe it's the way you set the JN to work?

@timmy5685  The behavior you’re observing is expected due to the presence of the Dev Tools plugin, which is enabled by default on all JN sites created from the GitHub link and forces sandbox/dev mode. If you disable the Dev Tools plugin, both the sandbox mode and the notification should disappear.

timmy5685 commented:
ok - thanks for confirming @mordeth!

So that is also what is causing the expired session issue?

dmallory42 commented:
> So that is also what is causing the expired session issue?

No, I don't think so - I'm looking into this one 🙂  do you remember if you left the page open for a while when you saw that error, by any chance?

dmallory42 committed:
Merge branch 'dev/i4-design-updates' of github.com:Automattic/woocommerce-payments into dev/i4-design-updates

dmallory42 committed:
Merge remote-tracking branch 'origin/develop' into dev/i4-design-updates

dmallory42 committed:
Remove uneeded changelog entries.

dmallory42 committed:
Update changelog.

dmallory42 committed:
remove unneeded change in merge commit.

dmallory42 committed:
Readd file lost in merge.

dmallory42 committed:
Update to use Woo purple.

dmallory42 commented:
Just to confirm, we merged a PR last week that should address any session issues. Please let us know if you spot anything like that again.

oaratovskyi commented:
> Just to confirm, we merged a PR last week that should address any session issues. Please let us know if you spot anything like that again.

cc @timmy5685

oaratovskyi commented:
@orcunomattic could you please take a look and review it design-wise? 🙏

oaratovskyi committed:
Merge branch 'develop' into dev/i4-design-updates

orcungogus commented:
@oaratovskyi, the _MOX + Stripe Embedded Onboarding_ screens look great!

My only comment (as I mentioned on Slack) is on a color update request for button, text link and inline text link elements used in the MOX flow, as well as the header top header we use for both MOX and Stripe Embedded Onboarding:
![Color](https://github.com/user-attachments/assets/cbde4af8-26f7-423b-b112-952bc1f4691e)
![color-update1](https://github.com/user-attachments/assets/4b038789-a82b-49dd-8744-c114088f356a)

The other color change request (if possible) I have is for the "Continue" CTA on the Stripe authentication pop-up: it was a blue button (in <Gutenberg Blueberry> color) previously, if we can revert it back to that WITHOUT affecting the top header area (which we switched to <Woo Purple 50>), that would be great, I am not sure if those two are connected:
![Color 2](https://github.com/user-attachments/assets/697d8097-358d-4b48-9ad0-99436915bc68)
![color-update2](https://github.com/user-attachments/assets/e7d0fb67-5bc6-41fd-a28d-22938e257b06)

oaratovskyi commented:
@orcunomattic thanks for the review!
About the change requests, let me share what I've explored:

> color update request for button, text link and inline text link elements used in the MOX flow, as well as the header top header we use for both MOX and Stripe Embedded Onboarding

The thing is that currently the order of the colors are next: 
- `--wp-components-color-accent` not set
- `--wp-admin-theme-color` Gutenberg Blue
- `#3858E9` Gutenberg blueberry
Our Embedded Stripe KYC ignores the theme color and is always Gutenberg Blueberry. 
I've updated the MOX to ignore the theme color too (image 1). For the focus of CTA button I used Gutenberg Blueberry 10% darker (image 2). Does it look good to you? 
![image](https://github.com/user-attachments/assets/95ffb817-7e5b-42bf-a62c-d500f308c5e9)
![image](https://github.com/user-attachments/assets/12a8fb35-9d5a-4612-974a-ea5f5cb73da2)

I thought about the consequence of this change: we have Gutenberg Blue CTA on the connect page and all other WooCommerce places and MOX + Stripe KYC having Gutenberg Blueberry. Could you confirm it's fine for now?
![image](https://github.com/user-attachments/assets/4441b64b-8bbe-426b-a6b2-cc98c25c8ebc)

> The other color change request (if possible) I have is for the "Continue" CTA on the Stripe authentication pop-up: it was a blue button (in color) previously, if we can revert it back to that WITHOUT affecting the top header area (which we switched to <Woo Purple 50>), that would be great, I am not sure if those two are connected

Unfortunately, they are connected. `primaryColor` variable sets both.

oaratovskyi committed:
Use Guternberg Blueberry on MOX

oaratovskyi committed:
Merge branch 'develop' into dev/i4-design-updates

oaratovskyi commented:
@timmy5685 As I've implemented @orcunomattic suggestions - merging this one and preparing the final build for you to test 🚀  
p.s. will wrangle any tweaks if needed in the follow-up PR

timmy5685 commented:
@orcunomattic for the CTA focus - I think it should be dark blueberry `#1D35B4`, right? Checking the style guide and I think that's what I'm seeing.

oaratovskyi commented:
> for the CTA focus - I think it should be dark blueberry #1D35B4, right? Checking the style guide and I think that's what I'm seeing.

@timmy5685 @orcunomattic I wrangled a PR to address this https://github.com/Automattic/woocommerce-payments/pull/9481
